### PR TITLE
Add coverage support

### PR DESCRIPTION
**Pull request by lmcgupe**
Recreated the pull request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,3 +15,23 @@ Theia is a cloud & desktop IDE implemented in TypeScript.
...
 You can find more details on how to run the examples [here](doc/Developing.md).
+ ### Running the electron example
+ ```
+ git clone https://github.com/TypeFox/Theia.git \
+ && cd Theia \
+ && npm install \
+ && cd examples \
+ && npm run install:theia \
+ && npm install \
+ && npm run cold:start:electron
```
Comment (svenefftinge) [inline README.md]: Did you add the electron description on purpose?

File: README.md [hunk 2]
```diff
@@ -15,3 +15,23 @@ Theia is a cloud & desktop IDE implemented in TypeScript.
...
+ ```
+### Running code coverage
+ ```
+npm run coverage
+
+### to view result
+
+firefox coverage/index.html
```
Comment (svenefftinge) [inline README.md]: Not sure this needs to be mentioned at all, but if this should go under doc/Developing.md

Comment (svenefftinge) [inline README.md]: Also does this work at all? There is no npm script called 'coverage' in the package.json

File: README.md (hunk 1 above)
Comment (kittaakos) [inline README.md]: These are redundant (and obsolete) steps. It is already available under the `Developing.md`.

**Comment by lmcgupe** [comment]
Ever since I rebased this coverage work, I get an error that I never got before. If you have an idea, please share. Thanks.
npm run coverage
theia@0.0.1-alpha.1 coverage /home/lmcgupe/Desktop/Theia
> nyc mocha --compilers ts:ts-node/register src/**/*.spec.ts


/home/lmcgupe/Desktop/Theia/node_modules/inversify/lib/annotation/decorator_utils.js:22
    if (Reflect.hasOwnMetadata(metadataKey, annotationTarget) === true) {
                ^
TypeError: Reflect.hasOwnMetadata is not a function
    at _tagParameterOrProperty (/home/lmcgupe/Desktop/Theia/node_modules/inversify/lib/annotation/decorator_utils.js:22:17)

**Comment by lmcgupe** [comment]
Ever since I rebased this coverage work, I get an error that I never got before. If you have an idea, please share. Thanks.
npm run coverage
theia@0.0.1-alpha.1 coverage /home/lmcgupe/Desktop/Theia
> nyc mocha --compilers ts:ts-node/register src/**/*.spec.ts


/home/lmcgupe/Desktop/Theia/node_modules/inversify/lib/annotation/decorator_utils.js:22
    if (Reflect.hasOwnMetadata(metadataKey, annotationTarget) === true) {
                ^
TypeError: Reflect.hasOwnMetadata is not a function
    at _tagParameterOrProperty (/home/lmcgupe/Desktop/Theia/node_modules/inversify/lib/annotation/decorator_utils.js:22:17)

**Comment by svenefftinge** [comment]
mocha needs to run with particular options. See https://github.com/theia-ide/theia/blob/master/config/mocha/mocha.opts

**Comment by lmcgupe** [comment]
ok got it.

**Comment by kittaakos** [comment]
Could we get rid of the `./node_modules/.bin/xyz/` from the `package.json` and use only `xyz` instead?

**Comment by lmcgupe** [comment]
comment adressed.

**Comment by lmcgupe** [comment]
good to merge ?

**Pull request merged**